compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -175,6 +190,7 @@ fn all_zero(buf: &[u8]) -> bool {
 impl<Prov: Provenance, Extra, Bytes, E: Encoder> Encodable<E> for Allocation<Prov, Extra, Bytes>
```

Comment:
Please just restrict this and the next instance to an allocation using `Box<[u8]>` for the bytes.